src/grt/src/fastroute/src/maze.cpp

```diff
@@ -174,6 +174,34 @@ void FastRouteCore::fixOverlappingEdge(
     routeLShape(
         startpoint, endpoint, blocked_positions, new_route_x, new_route_y);
 
+    // Updates the usage of the altered edge
+    const int edgeCost = nets_[net_id]->getEdgeCost();
+    for (int k = 0; k < treeedge->route.routelen;
+         k++) {  // remove the usages of the old edges
+      if (treeedge->route.gridsX[k] == treeedge->route.gridsX[k + 1]) {
+        if (treeedge->route.gridsY[k] != treeedge->route.gridsY[k + 1]) {
```

Comment:
Why this extra check here? Same for the code below.

Reply:
I made this based on another code, but i belive it is not necessary.

Reply:
what was the other code you based it on?

Reply:
It is based  on the code from the function removeLoops()

Reply:
Since loops should never happen in the first place I guess someone was just being extra careful and considering that you might also have zero length edges (which also shouldn't happen).  It isn't harmful so I we can leave it.

Reply:
Ok